lib/galaxy/util/compression_utils.py

```diff
@@ -47,8 +47,13 @@ def get_fileobj_raw(filename, mode="r", compressed_formats=None):
         compressed_format = 'bz2'
     elif 'zip' in compressed_formats and zipfile.is_zipfile(filename):
         # Return fileobj for the first file in a zip file.
-        with zipfile.ZipFile(filename, mode) as zh:
-            fh = zh.open(zh.namelist()[0], mode)
+        # 'b' is not allowed in the ZipFile mode argument
+        # since it always opens files in binary mode.
+        # For emulating text mode, we will be returning the binary fh in a
+        # TextIOWrapper.
+        zf_mode = mode.replace('b', '')
+        with zipfile.ZipFile(filename, zf_mode) as zh:
+            fh = zh.open(zh.namelist()[0], zf_mode)
```

Comment:
@wm75 Will this work with a zipped binary file? I'd have thought to use `mode` instead of `zf_mode`.

Reply:
Yes, the mode is always binary, no matter what you select here. The unit tests assert this.

Reply:
@nsoranzo surprised me too, but the `zh.open` does not accept 'b' in mode either and always returns a binary stream.

Reply:
@mvdbeek glad you suggested checking the stream type in the tests!